config/config_env.go

```diff
@@ -173,6 +183,28 @@ func reporterConfigFromEnv() (*ReporterConfig, error) {
 	// were not explicitly passed
 	rc.LocalAgentHostPort = fmt.Sprintf("%s:%d", host, port)
 
+	if e := os.Getenv(envEndpoint); e != "" {
+		u, err := url.ParseRequestURI(e)
+		if err != nil {
+			return nil, errors.Wrapf(err, "cannot parse env var %s=%s", envEndpoint, e)
+		}
+		rc.CollectorEndpoint = fmt.Sprintf("%s", u)
+	}
+
+	if e := os.Getenv(envUser); e != "" {
```

Comment:
can we use xor here instead and combine everything into 1 check? Albeit, it might not be a worthwhile optimization given this only happens during initialization but it's always fun to use xor every now and again.